### DISCUSSION TITLE
custom repo-server - where is the local cache kept?

### DISCUSSION THREAD

**Discussion by ekawas-td** [argoproj/argo-cd#9889] [Q&A] [answered]
We are using argocd **v2.3** and recently tried upgrading to **v2.4**

We built a custom repo-server image that copies across a script that is run on the repositories we had previously in the `/tmp` folder ...

The folder structure in the `/tmp` dir used to look like:

```
$ ls /tmp
helm183491347 helm231231232 https___github.com_XXX
```

Our custom image script used to look at the github subfolder and do a few things ... when we tested the update, it didn't look like our repo-server contained those subfolders anymore (aside from a bunch of GUID subfolders that were all empty).

Anyone have any ideas where to look for the repositories that are downloaded and what their naming convention might be now?

Thanks!

**Comment by crenshaw-dev** [comment, marked as answer]
As a security enhancement, all cached repos are now kept in `/tmp/_argocd-repo/{uuid}` where `{uuid}` is a random string (in uuid format).

The permissions of those directories are also locked down unless a repo operation is actively in progress. Your script will need to `chmod` the directories to restore permissions. Ideally it should `chmod` the permissions back to `0000` after it's finished.

**Reply by abdennour** [reply]
i like it. so it becomes 
```
d-wx------
```
if u wanna enable reading it `chmod a+r /tmp/_argocd-repo`
if u wanna restore as it was `chmod a-r /tmp/_argocd-repo`

**Reply by abdennour** [reply]
@crenshaw-dev is the random `{uuid}` is stored somewhere in the `kind: Application` resource, so at least,  we can MAP between this cache folder & its Application

**Comment by abdennour** [comment]
Hello, this feature introduced a regression bug with helm chart dependencies imported locally with local filepath `file://../xyz`

```yaml
# Chart.yaml
dependencies:
  - name: mychart
    repository: file://../mychart
    version: 0.1.0
   
```

**Reply by crenshaw-dev** [reply]
It's not a bug, it's intentional. A single repo- server may serve multiple tenants. Randomized paths prevent one tenant from accessing files belonging to another tenant.
